gormschema/gorm.go

```diff
@@ -187,3 +195,34 @@ func WithJoinTable(model any, field string, jointable any) Option {
 		})
 	}
 }
+
+type (
+	view interface {
+		ViewDef(*gorm.DB) gorm.ViewOption
+	}
+)
+
+// WithViews sets up callbacks to create views for the given "view-based" models.
+func WithViews(models ...any) Option {
+	return func(l *Loader) {
+		for _, model := range models {
+			if view, ok := model.(view); ok {
+				l.afterAutoMigrate = append(l.afterAutoMigrate, func(db *gorm.DB) error {
+					viewDef := view.ViewDef(db)
+					return db.Migrator().CreateView(inflect.Underscore(indirect(reflect.TypeOf(view)).Name()), gorm.ViewOption{
```

Comment:
Avoid stuff like this. Let the user defined the name.

Reply:
Fixed. I've used GORM's namer to make it compatible with GORM. By default, it will be named the same way as the GORM table, or it could be customized with TableName()